starlette/applications.py

```diff
@@ -140,16 +141,18 @@ def host(
     def add_middleware(
         self, middleware_class: type, **options: typing.Any
     ) -> None:  # pragma: no cover
+        if self.middleware_stack is not None:
```

Comment:
This method is deprecated, isn't it @Kludex ?

Reply:
Yes it is. #2002 still came up last meeting as a problem.

Reply:
It's not...

We didn't deprecate it. We only deprecated the decorators. :thinking: 

And "what came up on the meeting" was that I mentioned the `add_middleware` as an issue, due to the linked GitHub issue above, and the idea was to deprecate, and further remove, but Adrian came up with this alternative solution.

Reply:
Apologies, thank you for clarifying 😄